modules/cfe_testcase/src/es_info_test.c

```diff
@@ -0,0 +1,261 @@
+/*************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2019 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: es_info_test.c
+**
+** Purpose:
+**   Functional test of basic ES Information APIs
+**
+**   Demonstration of how to register and use the UT assert functions.
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+
+#include "cfe_test.h"
+
+void TestAppInfo(void)
+{
+    uint32           TestAppId;
+    uint32           ESAppId;
+    uint32           AppIdByName;
+    const char *     TestAppName = "TESTRUN_APP";
+    const char *     ESAppName   = "CFE_ES";
+    const char *     InvalidName = "INVALID_NAME";
+    char             AppNameBuf[OS_MAX_API_NAME + 4];
+    CFE_ES_AppInfo_t TestAppInfo;
+    CFE_ES_AppInfo_t ESAppInfo;
+
+    UtPrintf("Testing: CFE_ES_GetAppID, CFE_ES_GetAppIDByName, CFE_ES_GetAppName, CFE_ES_GetAppInfo");
+
+    UtAssert_INT32_EQ(CFE_ES_GetAppIDByName(&AppIdByName, TestAppName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetAppID(&TestAppId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(TestAppId, AppIdByName);
```

Comment:
Should not be tested as [u]int32, use CFE_RESOURCEID_TEST_EQUAL macro.

Reply:
@Alex is this fixed yet?

Reply:
doing it now, will be fixed soon.

Reply:
@jphickey Do we really need to use that because it changes a simple UtAssert_INT32_EQ(TestAppId, AppIdByName); to UtAssert_True(CFE_RESOURCEID_TEST_EQUAL(TestAppId, AppIdByName), "TestAppId (%d) == AppIdByName (%d)", TestAppId, AppIdByName);

Reply:
Yes, absolutely.   Code should never assume a value is a simple integer.  Importantly - IDs should not be interchanged/intermixed with other numbers or other ID types.

For instance it would be very easy in your tests (in fact you can try this!) to mistakenly call `CFE_ES_GetTaskID()` instead of `CFE_ES_GetAppID()` and/or assign an appID value to a taskID variable or something of that nature.  We need to make sure mistakes like this are caught - we've had bugs in the past because something like this was done.  If all values are just `uint32` then they cannot be caught, anything goes.

When you use `set(MISSION_RESOURCEID_MODE STRICT)` in your global_build_options.cmake file as I mentioned this type of mistake becomes a compile error.  The only reason its not set that way by default is for backward compatibility, but all framework devs should have this enabled.

The correct test should be something like: 
`UtAssert_True(CFE_RESOURCEID_TEST_EQUAL(TestAppId, AppIdByName), "TestAppId (%lu) == AppIdByName (%lu)", CFE_RESOURCEID_TO_ULONG(TestAppId), CFE_RESOURCEID_TO_ULONG(AppIdByName));`

Yep, its more verbose. but it is type-safe.